src/WidgetBase.ts

```diff
@@ -409,11 +437,22 @@ export class WidgetBase<P extends WidgetProperties = WidgetProperties, C extends
 			const render = beforeRenders.reduce((render, beforeRenderFunction) => {
 				return beforeRenderFunction.call(this, render, this._properties, this._children);
 			}, this.render.bind(this));
+
+			this._requiredNodes.forEach((element, key) => {
```

Comment:
In a separate PR I've already abstracting the beforeRenders and afterRenders into private functions.